storage/innobase/include/fil0crypt.h

```diff
@@ -327,6 +327,59 @@ struct fil_space_scrub_status_t {
   ulint current_scrub_max_page_number; /*!< current scrub max page no */
 };
 
+struct redo_log_key final {
+  uint version;
+  char key[ENCRYPTION_KEY_LEN];
+  ulint read_count;
+  ulint write_count;
+  bool present;
+
+  bool persisted() const noexcept { return version != 0; }
+};
+
+/** Handles the fetching/generation/storing/etc of keyring redo log keys.
+
+ This class is *NOT* thread safe, as thread safety is not required.
+ Data is only accessed/modified on the following points:
+ * When the redo space is created, at startup
+ * During redo log recovery, at startup
+ * When the server UUID is generated, at startup
+ * When the user requests a new key version, checked periodically in the
+   master thread
+
+ As these can't happen in parallel, no lock is used.
+ */
+class redo_log_keys final {
+ public:
+  /** Loads the latest redo log key from the keyring.
+   * @param[in]	generate If true, a key is generated if an existing key can't
+   * be loaded.
+   */
+  MY_NODISCARD
+  redo_log_key *load_latest_key(bool generate);
+  MY_NODISCARD
+  redo_log_key *load_key_version(uint version);
+
+  MY_NODISCARD
+  redo_log_key *generate_and_store_new_key();
+
+  /** These two methods are used during bootstrap encryption, when wo do not yet
+   * have an uuid */
```

Comment:
Also remove * here. And also comments with this format

Reply:
Done - but this is how clang-format formats comments unless you modify them by hand.

Reply:
clang-format adds * column to doxygen comments?

Reply:
If it breaks a long comment line (`ReflowComments: true`), it adds `*`s to all lines.

`clang-format-5.0`, it leaves short line alone, but newer versions also reformat blocks that do not break the line length limit - if MySQL ever moves to a more recent version, all of these comments will be reformatted by it.